src/colony/modules/net.js

```diff
@@ -538,10 +536,10 @@ TCPSocket.prototype.setNoDelay = function (val) {
   if (val) console.warn("Ignoring call to setNoDelay. TCP_NODELAY socket option not supported.");
 };
 
-function connect (port, host, callback, _secure) {
+function connect (options, callback, _secure) {
   var client = new TCPSocket(null, _secure);
   var args = Array.prototype.slice.call(arguments);
-  if (args.length === 4) args.pop();      // drop _secure param
+  if (args.length === 3) args.pop();      // drop _secure param
```

Comment:
Not following. If we popped the `args` array when there was only, say, one item in it…that would not be at all what we wanted.

Also, I wanted to leave this as "as-is" as possible since IIRC the proxy PR cleans this up a lot more aggressively.